### PR TITLE
express-notes-api - Marvin Loreto

### PR DESCRIPTION
**Pull request by marvinloreto**
GET_all
GET_all_status
POST_all_status
Post_err_500
DELETE_all_status
Delete_err_500
PUT_all_
Put_err_500

### CODE REVIEW

File: express-notes-api/index.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+  for (const property in dataJSON.notes) {
+    notesArray.push(dataJSON.notes[property]);
+  }
+  res.status(200).json(notesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = req.params.id;
```
Comment (BrettAlbright) [inline express-notes-api/index.js]: I would use the `Number` constructor here, to convert the string value to a number

File: express-notes-api/index.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  const note = dataJSON.notes[id];
+  if (id < 1) {
```
Comment (BrettAlbright) [inline express-notes-api/index.js]: you want to check specifically if id isNaN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN

You also want to confirm the number is an integer value: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger

File: express-notes-api/index.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        res.status(201).json(dataJSON.notes[id]);
+      }
+    });
+  }
+});
+
+app.delete('/api/notes/:id', (req, res) => {
+  const id = req.params.id;
+  const note = dataJSON.notes[id];
+  if (id < 1) {
```
Comment (BrettAlbright) [inline express-notes-api/index.js]: same as above, expand the check for isInteger and IsNaN for the id

File: express-notes-api/index.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        res.sendStatus(204);
+      }
+    });
+  }
+});
+
+app.put('/api/notes/:id', (req, res) => {
+  const id = req.params.id;
+  const updateNote = req.body;
+  if (id < 1) {
```
Comment (BrettAlbright) [inline express-notes-api/index.js]: same as above with the Number checks for is integer and is nan after converting the id to a Number

**Review by Zhiwutian — CHANGES_REQUESTED** [review]
Great work Marvin! Let me know if you have any questions on the feedback!

**Comment by marvinloreto** [comment]
Screenshot (247)
Screenshot (246)
Screenshot (244)

**Review by Zhiwutian — APPROVED** [review]
👍🏼

**Pull request merged**